SYCL/Regression/half_operators.cpp

```diff
@@ -69,8 +67,10 @@ template <typename T> bool test() {
 }
 
 int main(int argc, char **argv) {
+  sycl::queue queue{};
   bool passed = true;
-  passed &= test<float>();
-  passed &= test<sycl::half>();
+  passed &= test<float>(queue);
+  if (queue.get_device().has(sycl::aspect::fp16))
+    passed &= test<sycl::half>(queue);
```

Comment:
I believe the test compilation options need -fsycl-device-code-split=per_kernel to handle the aspects properly at runtime.

Reply:
Didn't know that, thanks!

Reply:
`per_kernel` device code split is not actually needed anymore since intel/llvm#7302: the compiler is now able to automatically split kernels into separate modules if optional features are used